crates/store/src/db/models/queries.rs

```diff
@@ -1072,7 +1071,155 @@ pub(crate) fn unconsumed_network_notes(
+//
+// Attention: uses the _implicit_ column `rowid`, which requires to use a few raw SQL nugget
+// statements
+#[allow(
+    clippy::cast_sign_loss,
+    reason = "We need custom SQL statements which has given types that we need to convert"
+)]
+#[allow(
+    clippy::too_many_lines,
+    reason = "Lines will be reduced when schema is updated to simplify logic"
+)]
+pub(crate) fn unconsumed_network_notes_for_account(
+    conn: &mut SqliteConnection,
+    network_account_id_prefix: NetworkAccountPrefix,
+    latest_block_num: BlockNumber,
+    mut page: Page,
+) -> Result<(Vec<NoteRecord>, Page), DatabaseError> {
+    assert_eq!(
+        NoteExecutionMode::Network as u8,
+        0,
+        "Hardcoded execution value must match query"
+    );
+
+    let rowid_sel = diesel::dsl::sql::<diesel::sql_types::BigInt>("notes.rowid");
+    let rowid_sel_ge =
+        diesel::dsl::sql::<diesel::sql_types::Bool>("notes.rowid >= ")
+            .bind::<diesel::sql_types::BigInt, i64>(page.token.unwrap_or_default() as i64);
+
+    // SELECT {}, rowid
+    // FROM notes
+    // LEFT JOIN note_scripts ON notes.script_root = note_scripts.script_root
+    // WHERE
+    //  execution_mode = 0 AND tag = ?1 AND
+    //  block_num <= ?2 AND
+    //  (consumed_block_num IS NULL OR consumed_block_num > ?2) AND rowid >= ?3
+    // ORDER BY rowid
+    // LIMIT ?
+    #[allow(
+        clippy::items_after_statements,
+        reason = "It's only relevant for a single call function"
+    )]
+    type RawLoadedTuple = (
+        i64,             // block_num
+        i32,             // batch_index
+        i32,             // note_index
+        Vec<u8>,         // note_id
+        i32,             // note_type
+        Vec<u8>,         // sender
+        i32,             // tag
+        i64,             // aux
+        i64,             // execution_hint
+        Vec<u8>,         // merkle_path
+        Option<Vec<u8>>, // assets
+        Option<Vec<u8>>, // inputs
+        Option<Vec<u8>>, // script
+        Option<Vec<u8>>, // serial_num
+        i64,             // rowid (from sql::<BigInt>("notes.rowid"))
+    );
+
+    #[allow(
+        clippy::items_after_statements,
+        reason = "It's only relevant for a single call function"
+    )]
+    fn split_into_raw_note_record_and_implicit_row_id(
+        tuple: RawLoadedTuple,
+    ) -> (NoteRecordRaw, i64) {
+        (
+            NoteRecordRaw {
+                block_num: tuple.0,
+                batch_index: tuple.1,
+                note_index: tuple.2,
+                note_id: tuple.3,
+                note_type: tuple.4,
+                sender: tuple.5,
+                tag: tuple.6,
+                aux: tuple.7,
+                execution_hint: tuple.8,
+                merkle_path: tuple.9,
+                assets: tuple.10,
+                inputs: tuple.11,
+                script: tuple.12,
+                serial_num: tuple.13,
+            },
+            tuple.14,
+        )
+    }
```

Comment:
Probably not for this PR, but this seems like a quite a bit of boilerplate code as compared to what we had before and would probably result in quite a bit of maintenance burden. Is there a way to simplify this?

Reply:
The root cause for the tuple stuff is the issue around the implicit row id which we use for pagination. The path forward is to change the `schema.rs` to explicitly contain the row id, or use a different way of pagination.

Reply:
PR addressing the issue (and a few more): https://github.com/0xMiden/miden-node/pull/1126